Drift/Private/DriftBase.cpp

```diff
@@ -782,49 +782,85 @@ void FDriftBase::LoadPlayerGameState(const FString& name, const FDriftGameStateL
     });
 }
 
+void FDriftBase::LoadPlayerGameState(int32 playerId, const FString& name, const FDriftGameStateLoadedDelegate& delegate)
+{
+    if (state_ != DriftSessionState::Connected)
+    {
+        DRIFT_LOG(Base, Warning, TEXT("Attempting to load player game state without being connected"));
+        delegate.ExecuteIfBound(ELoadPlayerGameStateResult::Error_InvalidState, name, FString());
+        onPlayerGameStateLoaded.Broadcast(ELoadPlayerGameStateResult::Error_InvalidState, name, FString());
```

Comment:
Is it useful to broadcast this (and the one on line 800)? It doesn't include which players gamestate you were trying to load so I'm skeptical if any listener can do anything with this event.

Reply:
Could do one of three things
1. Remove the calls
2. Add player Id to the delegate params
3. Create a new delegate for the player Id specific calls

@judgeaxl What do you think?

Reply:
I'd probably add the player id. If you want to see how we used these in the past, check Super Kaiju and The Machines. Pretty sure we use the passed in delegate over the global one, but perhaps a server which manages more states would be better off with the global one and an ID.